src/components/form/form.types.ts

```diff
@@ -282,3 +294,17 @@ export enum FormLayoutType {
      */
     Row = 'row',
 }
+
+/**
+ * @public
+ * Represents the JSON schema with Lime specific options
+ */
+export interface LimeJSONSchema extends JSONSchema7 {
```

Comment:
Yeah, I can understand that.
One idea I had that we could try is to create our own schema interface for what we use in our examples and maybe in the webclient and some customizations (the common ones).

And then also add like _unknowns_ like `any`
```typescript
export interface FormSchema {
    id?: string | undefined;
    $id?: string | undefined;
    
    /**
     * Lime elements specific options that can be specified in a schema
    */
    lime?: LimeSchemaOptions;
    type?: FormSchemaTypeName | FormSchemaTypeName[] | undefined;

    /* ... More properties/types ... */
    
    /* Allow whatever we haven't defined */
    [unknown: string]: any;
```

This way we get InteliSense for writing schemas, and it's easier to work with the FormComponents and so on.
BUT you can also set everything we haven't added yet.

Reply:
This would mean that the entire typing could look something like this:

```typescript

/**
 * @public
 * What type the schema represents
 */
type FormSchemaTypeName =
    | 'string'
    | 'number'
    | 'integer'
    | 'boolean'
    | 'object'
    | 'array'
    | 'null';

/**
 * @public
 * Utility type for representing a form object with recruisive types
 */
export interface FormObject {
    [key: string]: FormSchemaType;
}

/**
 * @public
 * Utility type for representing a form array with recruisive types
 */
export interface FormArray extends Array<FormSchemaType> {}

/**
 * @public
 * Available types in the schema
 */
export type FormSchemaType =
    | string
    | number
    | boolean
    | FormObject
    | FormArray
    | null;

/**
 * Meta schema
 *
 * Recommended values:
 * - 'http://json-schema.org/schema#'
 * - 'http://json-schema.org/hyper-schema#'
 * - 'http://json-schema.org/draft-07/schema#'
 * - 'http://json-schema.org/draft-07/hyper-schema#'
 *
 * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-5
 */
export type SchemaVersion = string;

/**
 * @public
 * Represents the schema to render a limel-form
 */
export interface FormSchema {
    id?: string | undefined;
    $id?: string | undefined;
    $ref?: string | undefined;
    $schema?: SchemaVersion | undefined;
    $comment?: string | undefined;

    /**
     * Lime elements specific options that can be specified in a schema
     */
    lime?: LimeSchemaOptions;

    /**
     * Handle all schema properties that are not explicitly defined
     */
    [unknown: string]: any;

    /**
     * @see https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-8.2.4
     * @see https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#appendix-A
     */
    $defs?:
        | {
              [key: string]: FormSchema;
          }
        | undefined;

    /**
     * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.1
     */
    type?: FormSchemaTypeName | FormSchemaTypeName[] | undefined;
    enum?: FormSchemaType[] | undefined;
    const?: FormSchemaType | undefined;

    /**
     * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.3
     */
    maxLength?: number | undefined;
    minLength?: number | undefined;
    pattern?: string | undefined;

    /**
     * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.4
     */
    items?: FormSchema | FormSchema[] | undefined;
    additionalItems?: FormSchema | boolean | undefined;
    uniqueItems?: boolean | undefined;
    contains?: FormSchema | undefined;

    /**
     * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.5
     */
    required?: string[] | undefined;
    properties?:
        | {
              [key: string]: FormSchema;
          }
        | undefined;
    additionalProperties?: FormSchema | boolean | undefined;
    dependencies?:
        | {
              [key: string]: FormSchema | string[];
          }
        | undefined;

    /**
     * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-6.7
     */
    allOf?: FormSchema[] | undefined;
    anyOf?: FormSchema[] | undefined;
    oneOf?: FormSchema[] | undefined;
    not?: FormSchema | undefined;

    /**
     * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-7
     */
    format?: string | undefined;

    /**
     * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-9
     */
    definitions?:
        | {
              [key: string]: FormSchema;
          }
        | undefined;

    /**
     * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-10
     */
    title?: string | undefined;
    description?: string | undefined;
    default?: FormSchemaType | undefined;
    readOnly?: boolean | undefined;
}

```

Reply:
I think it would be quite safe to expose this one. They seem to create a new interface for each version of the JSON schema specification, so risk of breaking changes are probably quite slim

Reply:
Ok, so would you prefer the extending of `JSONSchema7` instead of writing our own simpler implementation?

Reply:
Now we went with the extension of `JSONSchema7`. Seems to work for all our examples.